ext/DFTKManifoldsManoptRATExt.jl

```diff
@@ -0,0 +1,503 @@
+* `ρ=guess_density(basis), # would be consistent with other scf solvers
+* `tol=1e-6`: stop when the change in the density is less than this tolerance. If you set the `stopping_criterion=` directly, this keyword has no effect.
+    If you set the `stopping_criterion=` directly, this keyword has no effect.
+* `cost=nothing`: provide an individual cost function. You then also have to provide `gradient=`
+* `gradient=nothing`: provide an individual gradient function. You then also have to provide `cost=`
+    If you provide both of these, for the access to `:Energies`, `:Hamiltonian`, `:HamiltonianEvaluation`, and `:ρ`
+    you also have to implement `Manopt.get_parameter` for these.
+* `evaluation`=[`InplaceEvaluation`](@extref `Manopt.InplaceEvaluation`) whether the gradient of the objective is allocating or in-place.,
+* `manifold=`[`Stiefel`](@extref `Manifolds.Stiefel`): the manifold the optimisation is running on.
+  The default cost function allows to also use [`Grassmann`](@extref `Manifolds.Grassmann`).
+    If you set the `manifold_constructor=` directly, e.g. to switch to a real manifold, this keyword is ignored
+* `preconditioner=`[`PreconditionerTPA`](@ref)` a preconditioner to use for the Newton equation or the gradient.
+* `manifold_constructor=(n, k) -> manifold(n, k, ℂ)` the complete constructor for a single manifold within the
+     product manifold the optimization is defined on.
+* `stopping_criterion=[`StopAfterIteration`](@extref `Manopt.StopAfterIteration`)` `[`|`](@extref Manopt.StopWhenAny)` `[`StopWhenDensityChangeLess`](@ref)`(tol,deepcopy(ρ))`:
+    a stopping criterion for the algorithm when to stop. Uses `maxiter=` and `tol=` as defaults, requires that `ρ` a density.
+* `record=[[`RecordCost`](@extref `Manopt.RecordCost`)`() => :Etot, `[`RecordDensityChange`](@ref)`(ρ) => :Δρ`, `[`RecordTime`](@extref `Manopt.RecordTime`)`(; mode=:total) => :time]`
+    specify values to record diring the iterations, where in a `pair` the second determines how to access the values; the three default ones are included in the returned named tuple
+* `retraction_method = `[`ProjectionRetraction`](@extref `Manifolds.ProjectionRetraction`) the retration to use on the manifold to more into a certain direction.
+* `vector_transport_method=`[`ProjectionTransport`](@extref `Manifolds.ProjectionTransport`)`()` the vector transport to use to move tangent vectors between tangent spaces.
+* `stepsize = `[`ArmijoLinesearch`](@extref `Manopt.ArmijoLinesearch`)`(; retraction_method=retraction_method)`
+   specify a step size rule.
+
+All other keyword arguments are passed to the solver state constructor as well as to
+both a decorate for the objective and the state.
+This allows both to set other setting for a solver but also to add `debug=` funtionality
+or even a `cache=` to the objective, though the default objective already caches parts
+that both a cost and a gradient at the same point would require.
```

Comment:
I would add here a note that this interface is not yet stable and may change without warning, because it's quite low-level and I want to reserve ourselves the right to have breaking changes in the future.

Reply:
Sure. Did that.